examples/wallet/app/components/RestoreWalletFromMnemonic.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+import React, { useState } from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import typeof { setAccountFromMnemonic as SetAccountFromMnemonic } from '../actions/account';
+import { isValidMnemonic } from '../utils/mnemonic';
+import CreateUnlockWalletPassword from '../containers/CreateUnlockWalletPassword';
+
+type Props = {
+  setAccountFromMnemonic: SetAccountFromMnemonic
+};
+
+export default ({ setAccountFromMnemonic }: Props) => {
+  const checkIsValidMnemonicPhrase = function checkIsValidMnemonicPhrase() {
+    setIsMnemonicValid(isValidMnemonic(newMnemonicPhrase));
+  };
+
+  const handleSubmitMnemonic = function handleSubmitMnemonic(event) {
+    event.preventDefault();
+    if (isValidMnemonic(newMnemonicPhrase)) {
+      // TODO: ADD IF SENTENCE TO CHECK IF PASSWORD AND CONFIRM IS OK
+      if (isValidUnlockPassword) {
+        return setAccountFromMnemonic(
+          newMnemonicPhrase,
+          newMnemonicPassword,
+          unlockWalletPassword
+        );
+      }
+    }
+    setIsMnemonicValid(false);
+  };
+
+  const setValidCreateUnlockWalletPassword = function setValidCreateUnlockWalletPassword(
+    unlockPwd: string,
```

Comment:
you could export the type of the callback in the called component and import it wherever it's used, to avoid having to type the handlers manuallly every time